### PR TITLE
Merge sort

### PR DESCRIPTION
**Pull request by jonathanstallings**

### CODE REVIEW

File: merge_sort.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        while x < len(left_half):
+            un_list[z] = left_half[x]
+            x += 1
+            z += 1
+
+        while y < len(right_half):
+            un_list[z] = right_half[y]
+            y += 1
+            z += 1
```
Comment (cewing) [inline merge_sort.py]: good.  Can you break the merging part into it's own stand-alone function?

File: merge_sort.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    from timeit import Timer
+
+    SETUP = """from __main__ import BEST_CASE, WORST_CASE, merge_srt"""
+
+    best = Timer('merge_srt({})'.format(BEST_CASE), SETUP).timeit(1000)
+
+    worst = Timer('merge_srt({})'.format(WORST_CASE), SETUP).timeit(1000)
+
+    print("""Best case represented as a list that is already sorted\n
+        Worst case represented as a list that is absolute reverse of sorted""")
```
Comment (cewing) [inline merge_sort.py]: that doesn't look to me like what you have written above as the worst case list.  Is this description accurate, or is it copied from elsewhere.  Also, newlines are respected in Python triple-quoted strings, so you don't need to add a `\n` char to yours.

File: test_merge_sort.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+from merge_sort import merge_srt
+
+
+def test_merge_sort():
+    expected = range(20)
+    unsorted = expected[:]
+    shuffle(unsorted)
+    merge_srt(unsorted)
+    actual = unsorted
```
Comment (cewing) [inline test_merge_sort.py]: why not bind `actual` to the copy of `expected` in the first place and avoid the need to rebind after the shuffle-sort dance?

File: merge_sort.py (hunk 1 above)
Comment (jonathanstallings) [inline merge_sort.py]: Good idea.

File: merge_sort.py (hunk 2 above)
Comment (jonathanstallings) [inline merge_sort.py]: Good points. Indeed. We were mistaken about the worst case. I believe it should worst case should be a random list.

File: test_merge_sort.py (hunk 3 above)
Comment (jonathanstallings) [inline test_merge_sort.py]: Makes sense. I think the rebind here was just for readability sake, but, upon reflection, it is unnecessary.

**Pull request merged**